File: scratch/do_git_push.js
const { execSync } = require('child_process');
const fs = require('fs');
const path = require('path');

const logFile = path.join(__dirname, 'git_result.txt');
fs.writeFileSync(logFile, '=== STARTING GIT PUSH SCRIPT ===\n', 'utf8');

function log(msg) {
  fs.appendFileSync(logFile, msg + '\n', 'utf8');
}

const commands = [
  'git status',
  'git add .',
  'git commit -m "Feat: Them cau hoi keo tha, noi cau va phan quyen admin xem bai hoc da khoa"',
  'git push origin main',
  'git push'
];

for (const cmd of commands) {
  log(`\n=== Running: ${cmd} ===`);
  try {
    const stdout = execSync(cmd, { 
      cwd: path.resolve(__dirname, '..'),
      encoding: 'utf8',
      timeout: 20000,
      env: { ...process.env, GIT_TERMINAL_PROMPT: '0' }
    });
    log(`STDOUT:\n${stdout}`);
  } catch (err) {
    log(`ERROR/STDERR:\n${err.message}\n${err.stdout || ''}\n${err.stderr || ''}`);
  }
}

log('\n=== COMPLETED GIT PUSH SCRIPT ===');

File: scratch/do_git_push.py
import subprocess
import os

log_file = r"d:\OneDrive\2025 - 2026\Hoc tap Lop 5\scratch\git_result.txt"

def log(msg):
    with open(log_file, "a", encoding="utf-8") as f:
        f.write(msg + "\n")

def run_git():
    if os.path.exists(log_file):
        os.remove(log_file)
        
    repo_dir = r"d:\OneDrive\2025 - 2026\Hoc tap Lop 5"
    
    commands = [
        "git status",
        "git add .",
        'git commit -m "Feat: Them cau hoi keo tha, noi cau va phan quyen admin xem bai hoc da khoa"',
        "git push origin main",
        "git push"
    ]
    
    for cmd in commands:
        log(f"=== Running: {cmd} ===")
        try:
            res = subprocess.run(cmd, cwd=repo_dir, shell=True, capture_output=True, text=True, timeout=15)
            log(f"STDOUT:\n{res.stdout}")
            if res.stderr:
                log(f"STDERR:\n{res.stderr}")
            log(f"RETURNCODE: {res.returncode}\n")
        except Exception as e:
            log(f"ERROR / TIMEOUT: {str(e)}\n")

if __name__ == "__main__":
    run_git()
